modules/infra/submodules/network/vpc.tf

```diff
@@ -49,6 +49,57 @@ resource "aws_vpc_endpoint" "s3" {
   }
 }
 
+data "aws_prefix_list" "s3" {
+  prefix_list_id = aws_vpc_endpoint.s3[0].prefix_list_id
+}
+
+resource "aws_security_group" "ecr_endpoint" {
+  name        = "${var.deploy_id}-ecr"
+  description = "ECR Endpoint security group"
+  vpc_id      = aws_vpc.this[0].id
+
+  lifecycle {
+    create_before_destroy = true
+  }
+  tags = {
+    "Name" = "${var.deploy_id}-ecr"
+  }
+}
+
+resource "aws_vpc_endpoint" "ecr_dkr" {
```

Comment:
Should we also make these generally optional? That way we/customers don't need to pay for vpc endpoints when the outbound restriction isn't being used?

Reply:
Done!
added a `create_ecr_endpoint` boolean flag in the network section that enables the feature

Reply:
Cool. We will need to make sure the platform-apps side makes use of this toggle as well.